### PR TITLE
Add communication channel to interaction

### PR DESCRIPTION
**Pull request by ztolley**
- Adds the 'communication_channel' to the interaction form
- Removes the use of interaction type when posting an interaction
- No longer fetches interaction types for the edit form

### CODE REVIEW

File: src/apps/interactions/transformers.js [hunk 1]
```diff
@@ -129,11 +129,10 @@ function transformInteractionResponseToViewRecord ({
   return viewRecord
 }
 
-function transformInteractionFormBodyToApiRequest ({ props, company, communicationChannel }) {
+function transformInteractionFormBodyToApiRequest ({ props, company }) {
   return assign({}, props, {
     date: transformDateObjectToDateString('date')(props),
-    company: company,
-    communication_channel: communicationChannel,
+    company,
```
Comment (teneightfive) [inline src/apps/interactions/transformers.js]: Move up one line? Easier to read when shorthand declarations are first.

File: src/apps/interactions/macros.js [hunk 2]
```diff
@@ -120,6 +120,15 @@ const interactionEditFormConfig = function ({
           return advisers.map(transformContactToOption)
         },
       },
+      {
+        macroName: 'MultipleChoiceField',
+        name: 'communication_channel',
+        label: formLabels.communication_channel,
```
Comment (feedmypixel) [inline src/apps/interactions/macros.js]: we can just add labels for all in a map like [here](https://github.com/uktrade/data-hub-frontend/blob/b3f8c3afe960cdfa13adb71f978ce2139198aaa4/src/apps/events/macros/event-filters-fields.js#L48)?

File: src/apps/interactions/middleware/details.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const { assign, find } = require('lodash')
+const { assign } = require('lodash')
```
Comment (feedmypixel) [inline src/apps/interactions/middleware/details.js]: This is something we have elsewhere, can we keep a hard return here to define between external requires and internal ones.

File: src/apps/interactions/macros.js (hunk 2 above)
Comment (ztolley) [reply src/apps/interactions/macros.js]: was trying to keep the pr small and focussed just on adding this field.

**Pull request merged**